docs/design.md

```diff
@@ -91,97 +107,204 @@ This pattern is inspired by [sickle_ui](https://github.com/UmbraLuminosa/sickle_
 
 ```rust
-app.register_type::<HandleMap<SpriteKey>>();
-app.init_resource::<HandleMap<SpriteKey>>();
+fn all_assets_loaded(
+    image_handles: Res<ImageHandles>,
+) -> bool {
+    image_handles.all_loaded(&asset_server)
+}
 ```
 
 ### Reasoning
 
 This pattern is inspired by [bevy_asset_loader](https://github.com/NiklasEi/bevy_asset_loader).
 By preloading your assets, you can avoid hitches during gameplay.
+We start loading as soon as the app starts and wait for all assets to be loaded in the loading screen.
 
-Using an enum to represent your assets encapsulates their file path from the rest of the game code,
-and gives you access to static tooling like renaming in an IDE, and compile errors for an invalid name.
+By using strings as keys, you can dynamically load assets based on input data such as a level file.
+If you prefer a purely static approach, you can also use an `enum YourAssetHandleKey` and `impl AsRef<str> for YourAssetHandleKey`.
+You can also mix the dynamic and static approach according to your needs.
 
-## Spawning
+## Spawn Command
 
 ### Pattern
 
-Spawn a game object by using an observer:
+Spawn a game object by using a custom command. Inside the command,
+run the spawning code with `world.run_system_once` or  `world.run_system_once_with`:
 
 ```rust
 // monster.rs
-use bevy::prelude::*;
 
-pub(super) fn plugin(app: &mut App) {
-    app.observe(on_spawn_monster);
+#[derive(Debug)]
+pub struct SpawnMonster {
+    pub health: u32,
+    pub transform: Transform,
 }
 
-#[derive(Event, Debug)]
-pub struct SpawnMonster;
+impl Command for SpawnMonster {
+    fn apply(self, world: &mut World) {
+        world.run_system_once_with(self, spawn_monster);
+    }
+}
 
-fn on_spawn_monster(
-    _trigger: Trigger<SpawnMonster>,
+fn spawn_monster(
+    spawn_monster: In<SpawnMonster>,
     mut commands: Commands,
 ) {
     commands.spawn((
         Name::new("Monster"),
+        Health::new(spawn_monster.health),
+        SpatialBundle::from_transform(spawn_monster.transform),
         // other components
     ));
 }
 ```
 
-And then, somewhere else in your code, trigger the observer:
+And then, somewhere else in your code, add the command to the command queue:
 
 ```rust
+// dangerous_forest.rs
+
 fn spawn_monster(mut commands: Commands) {
-    commands.trigger(SpawnMonster);
+    commands.add(SpawnMonster {
+        health: 100,
+        transform: Transform::from_xyz(10.0, 0.0, 0.0),
+    });
 }
 ```
 
 ### Reasoning
 
 By encapsulating the spawning of a game object in a function,
 you save on boilerplate code and can easily change the behavior of spawning.
-An observer is an elegant way to then trigger this function from anywhere in your code.
-A limitation of this approach is that calling code cannot extend the spawn call with additional components or children.
-If you know about a better pattern, please let us know!
+A custom command is an elegant way to then indirectly call this function from anywhere in your code.
```

Comment:
```suggestion
Calling a system within the command makes it easy to add system parameters to access
things like assets and resources while spawning the entity.
```

Reply:
Not sure about this one. The new version reads more confusing to me.

Reply:
The previous two lines make this line redundant. Better to replace it with a line explaining the reasoning behind the `run_system_once_with` call.

Reply:
Fair enough. How about this: [`63de369` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/63de3699ef54a9de2dc714e5b46005050ce48681)?